packages/rolldown/src/cli/arguments/index.ts

```diff
@@ -43,14 +48,23 @@ export function parseCliArguments() {
     options,
     tokens: true,
     allowPositionals: true,
-    allowNegative: true,
     // We can't use `strict` mode because we should handle the default config file name.
     strict: false,
   })
 
   tokens
     .filter((token) => token.kind === 'option')
     .forEach((option) => {
+      if (option.name.startsWith('no-')) {
```

Comment:
I'm not sure about this `no-xxx` patterns. We don't need to follow what rollup did in cli.

I'll suggest we made decisions based on implementing a cli with the best UX, not just simply rollup did it.

Reply:
- `rolldown --treeshake`
- `rolldown --treeshake=false` | `rolldown --no-treeshake`
- `rolldown --treeshake=false --external-live-bindings=false`  | `rolldown --no-treeshake --no-external-live-bindings`

Reply:
There's usage that using rolldown-cli programmatically. `no-xxx` pattern is not friendly for rendering cli args programmatically.

```js
const command = `rolldown --treeshake=${treeshake}`
```

```js
const command = `rolldown --${ !treeshake ? 'no-'  : ''}treeshake`
```

Reply:
I have implemented the no- name prefix to handle the boolean value whose default values are true. The only problem is that a union type with string and boolean is hard to handle.

Reply:
I think we can follow up Unix's UX Pattern since it’s easier to support and more popular to use.

For arguments,

1. Offer both short and long options where applicable. For example, `-o` and `--output`.
2. Use the presence of an option to toggle a feature. For disabling, use `--no-` prefixes for long options.
3. Allow flexibility in how users pass values. Support both space-separated (for short option) and equals-sign (for long option) syntax.
4. Clearly define mandatory and optional positional arguments, and ensure options can be used flexibly with them.
5. Support stacked short options for a concise CLI experience.
6. Always provide a clear and consistent way to negate options where applicable.
7. Always include `--help` and `--version` as standard options.
8. Support environment variables for configurations, with CLI options taking precedence.
9. Use defaults where possible but ensure clear communication when required arguments are omitted.
10. Support configuration files for flexibility but ensure CLI arguments can override any configuration.

*(Generated by ChatGPT)*

Reply:
> I'm not sure about this no-xxx patterns. We don't need to follow what rollup did in cli.

Incidentally, the code responsible for handling arguments prefixed with “no-“ is referenced from the official website of Node.js: [node.js official website](https://nodejs.org/api/util.html#parseargs-tokens).